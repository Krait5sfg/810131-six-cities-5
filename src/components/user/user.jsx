import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {AuthorizationStatus} from '../../utils/const';

const User = ({onLinkEmailClick, authorizationStatus}) => {
  return (
    <li className="header__nav-item user">
      <a className="header__nav-link header__nav-link--profile" href="#" onClick={onLinkEmailClick}>
        <div className="header__avatar-wrapper user__avatar-wrapper">
        </div>

        {authorizationStatus === AuthorizationStatus.NO_AUTH
          ? <span className="header__login">Sign in</span>
          : <span className="header__user-name user__name">Oliver.conner@gmail.com</span>
        }

      </a>
    </li>
  );
};

User.propTypes = {
  onLinkEmailClick: PropTypes.func.isRequired,
  authorizationStatus: PropTypes.string.isRequired
};

const mapStateToProps = (({USER}) => ({
  authorizationStatus: USER.authorizationStatus,
}));

export {User};
export default connect(mapStateToProps)(User);
