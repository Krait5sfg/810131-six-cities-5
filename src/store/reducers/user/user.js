import {AuthorizationStatus} from '../../../utils/const';
import {ActionType} from '../../../store/action';

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  user: {},
};

// отвечает за авторизацию
const user = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return Object.assign({}, state, {authorizationStatus: action.payload});
    case ActionType.UPDATE_USER:
      return Object.assign({}, state, {user: action.payload});
  }
  return state;
};

export {user};
