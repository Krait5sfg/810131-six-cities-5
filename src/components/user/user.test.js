import React from 'react';
import renderer from 'react-test-renderer';
import {User} from './user';

const userData = {
  avatarUrl: ``,
  email: `test@mail.ru`,
  id: 1,
  isPro: false,
  name: `test`,
};

describe(`User render correctly`, () => {
  it(`User with authorization`, () => {
    const tree = renderer
      .create(<User
        onLinkEmailClick={() => {}}
        authorizationStatus={`AUTH`}
        userData={userData}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`User without authorization`, () => {
    const tree = renderer
      .create(<User
        onLinkEmailClick={() => {}}
        authorizationStatus={`NO_AUTH`}
        userData={{}}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
