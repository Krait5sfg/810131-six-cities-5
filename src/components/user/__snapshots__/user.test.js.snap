// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User render correctly User with authorization 1`] = `
<li
  className="header__nav-item user"
>
  <a
    className="header__nav-link header__nav-link--profile"
    href="#"
    onClick={[Function]}
  >
    <div
      className="header__avatar-wrapper user__avatar-wrapper"
    />
    <span
      className="header__user-name user__name"
    >
      test@mail.ru
    </span>
  </a>
</li>
`;

exports[`User render correctly User without authorization 1`] = `
<li
  className="header__nav-item user"
>
  <a
    className="header__nav-link header__nav-link--profile"
    href="#"
    onClick={[Function]}
  >
    <div
      className="header__avatar-wrapper user__avatar-wrapper"
    />
    <span
      className="header__login"
    >
      Sign in
    </span>
  </a>
</li>
`;
